src/backend/api/meals.js

```diff
@@ -1,15 +1,92 @@
 const express = require("express");
-const router = express.Router();
-const knex = require("../database");
+const mealsRouter = express.Router();
+const dbcon = require("../database");
 
-router.get("/", async (request, response) => {
+// get all meals
+mealsRouter.get("/", async (request, response) => {
+  let meals = undefined;
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("Meal").select("title");
-    response.json(titles);
+    meals = await dbcon("Meal");
   } catch (error) {
-    throw error;
+    return response.status(400).send(error);
   }
+  if (meals.length < 1) {
+    return res.status(204).send();
+  }
+  response.send(meals);
+});
+
+// post a meal
+mealsRouter.post("/", async (request, response) => {
+  const meal = request.body;
+  try {
+    await dbcon("Meal").insert(meal);
+  } catch (error) {
+    return response.status(400).send(error);
+  }
+  response.send("ok");
+});
+
+// get a meal by id
+mealsRouter.get("/:id", async (request, response) => {
+  let meal = undefined;
+  try {
+    const { id } = request.params;
+    [meal] = await dbcon("Meal").where({ id: id });
+  } catch (error) {
+    return response.status(400).send();
+  }
+  if (!meal) {
+    return response.status(204).send();
+  }
+  response.send(meal);
 });
 
-module.exports = router;
+// update a meal by id
+mealsRouter.put("/:id", async (request, response) => {
+  try {
+    const [meal] = await dbcon("Meal").where({ id: request.params.id });
+    if (!meal) {
+      return response.status(204).json({
+        success: false,
+        message: "Meal with the given Id doesn't exist!",
+      });
+    } else {
+      try {
+        const updateInfo = request.body;
+        await dbcon("Meal").where({ id: request.params.id }).update(updateInfo);
+        response.json({
+          success: true,
+          message: "Meal with the given Id has been updated",
+        });
+      } catch (err) {
+        return response.status(400).send(err);
```

Comment:
400 would be often for bad request so when you verify request body or parameter. In this case I would use 500 in this case. And you could merge two try catch into one try/catch